### PR TITLE
MM-21389: add godocs to most of the codebase

### PR DESCRIPTION
**Pull request by agnivade**
#### Ticket Link

https://mattermost.atlassian.net/browse/MM-21389

### CODE REVIEW

**Comment by agnivade** [comment]
I have left out the big interface APIs and the config package. They can be done later in small chunks.

**Comment by streamer45** [comment]
> I have left out the big interface APIs and the config package. They can be done later in small chunks.

Sounds good. Great work :+1: . Will review it shortly.

File: loadtest/control/simplecontroller/controller.go [hunk 1]
```diff
@@ -33,6 +35,8 @@ func New(id int, user user.User, status chan<- control.UserStatus) *SimpleContro
 	}
 }
 
+// Run begins performing a set of actions in a loop with a defined wait
+// in between the actions. It keeps on doing it until Stop is invoked.
```
Comment (streamer45) [inline loadtest/control/simplecontroller/controller.go]: I would only add that this is a blocking function and it's meant to be used inside a goroutine.

File: loadtest/control/simplecontroller/actions.go [hunk 2]
```diff
@@ -12,6 +12,8 @@ import (
 	"github.com/mattermost/mattermost-server/v5/model"
 )
 
+// UserAction specifies an action performed by a user
+// followed by a wait time.
 type UserAction struct {
```
[the remark this answers is not in the document]
Comment (agnivade) [reply loadtest/control/simplecontroller/actions.go]: I will remove the godoc, and we can un-export this in another PR.

**Pull request merged**